cmake/Modules/FindMKL.cmake

```diff
@@ -88,25 +88,41 @@ ELSE ("${SIZE_OF_VOIDP}" EQUAL 8)
   SET(iccvers "ia32")
   SET(mkl64s)
 ENDIF ("${SIZE_OF_VOIDP}" EQUAL 8)
-IF(CMAKE_COMPILER_IS_GNUCC)
-  IF ("${MKL_THREADING}" STREQUAL "TBB")
-    SET(mklthreads "mkl_tbb_thread")
-    SET(mklrtls "tbb")
-  ELSE()
-    SET(mklthreads "mkl_gnu_thread" "mkl_intel_thread")
-    SET(mklrtls "gomp" "iomp5")
-  ENDIF()
-  SET(mklifaces  "intel" "gf")
-ELSE(CMAKE_COMPILER_IS_GNUCC)
+
+IF(WIN32)
   IF ("${MKL_THREADING}" STREQUAL "TBB")
     SET(mklthreads "mkl_tbb_thread")
-    SET(mklrtls "tbb")
+    IF (CMAKE_BUILD_TYPE STREQUAL Debug)
+      SET(mklrtls "tbb12_debug")
+    ELSE ()
+      SET(mklrtls "tbb12")
+    ENDIF ()
   ELSE()
     SET(mklthreads "mkl_intel_thread")
-    SET(mklrtls "iomp5" "guide")
+    SET(mklrtls "libiomp5md")
   ENDIF()
-  SET(mklifaces  "intel")
-ENDIF (CMAKE_COMPILER_IS_GNUCC)
+    SET(mklifaces  "intel")
+ELSE(WIN32)
+  IF(CMAKE_COMPILER_IS_GNUCC)
+    IF ("${MKL_THREADING}" STREQUAL "TBB")
+      SET(mklthreads "mkl_tbb_thread")
+      SET(mklrtls "tbb")
+    ELSE()
+      SET(mklthreads "mkl_gnu_thread" "mkl_intel_thread")
+      SET(mklrtls "gomp" "iomp5")
+    ENDIF()
+    SET(mklifaces  "intel" "gf")
+  ELSE(CMAKE_COMPILER_IS_GNUCC)
+    IF ("${MKL_THREADING}" STREQUAL "TBB")
+      SET(mklthreads "mkl_tbb_thread")
+      SET(mklrtls "tbb")
+    ELSE()
+      SET(mklthreads "mkl_intel_thread")
+      SET(mklrtls "iomp5" "guide")
+    ENDIF()
+    SET(mklifaces  "intel")
+  ENDIF (CMAKE_COMPILER_IS_GNUCC)
```

Comment:
Can you please explain why one needs section that differentiates between gcc and clang?

Reply:
It is original code, I don't known why it. I only add a section for `Win32` and move the origianl code to `else` block.